components/apimgt/org.wso2.carbon.apimgt.common.analytics/src/main/java/org/wso2/carbon/apimgt/common/analytics/publishers/impl/AbstractRequestDataPublisher.java

```diff
@@ -43,27 +45,40 @@ public abstract class AbstractRequestDataPublisher implements RequestDataPublish
     @Override
     public void publish(Event analyticsEvent) {
 
-        CounterMetric counterMetric = this.getCounterMetric();
-        if (counterMetric == null) {
-            log.error("counterMetric cannot be null.");
+        Map<String, Object> dataMap = OBJECT_MAPPER.convertValue(analyticsEvent, MAP_TYPE_REFERENCE);
+        List<CounterMetric> multipleCounterMetrics = this.getMultipleCounterMetrics();
+        if (multipleCounterMetrics == null) {
+            log.error("All the counterMetrics are invalid. Event will be dropped.");
             return;
         }
 
-        Map<String, Object> dataMap = OBJECT_MAPPER.convertValue(analyticsEvent, MAP_TYPE_REFERENCE);
-        MetricEventBuilder builder = counterMetric.getEventBuilder();
-        for (Map.Entry<String, Object> entry : dataMap.entrySet()) {
-            try {
-                builder.addAttribute(entry.getKey(), entry.getValue());
-            } catch (MetricReportingException e) {
-                log.error("Error adding data to the event stream.", e);
-                return;
+        for (CounterMetric counterMetric : multipleCounterMetrics) {
+            if (counterMetric == null) {
+                log.error("counterMetric cannot be null.");
+            } else {
+                String counterMetricClassName = counterMetric.getClass().toString().
+                        replaceAll("[\r\n]", "").split(" ")[1];
+                log.info("Started adding data to counterMetric " + counterMetricClassName);
```

Comment:
Won't this log print for each and every event?

Reply:
Yes it will Anusha

Reply:
Since we are logging in case of failure, there is no need to log success explicitly. So i am removing this log